### ISSUE TITLE
RaGOO out visualization

### ISSUE THREAD

**Issue by Ural-Yunusbaev** [malonge/RaGOO#40]
Hi Michael! 

Thank you for the super-fast RaGOO algorithm!
Do you have any good solutions for fast RaGOO out visualization?
Do you have any idea how to visualize 
contigs_against_ref.paf
and the files from /groupings /orderings?
Thank you in advance!

Best regards,
Ural

**Comment by malonge** [comment]
Hi there,

Were you thinking something along the lines of a dotplot like mummerplot or assemblytics? Those both require base level alignments so the contigs_against_ref.paf can't be used for those. Dotplots cant be easily made in R or python, but the tricky part is displaying the many contigs against the many reference chromosomes. 

Or maybe you had some other visualization in mind?

**Comment by mictadlo** [comment]
Hi,
I would be interested in visualising structural variation from Ragoo's output.

Thank you in advance,

Michal

**Comment by Ural-Yunusbaev** [comment]
Yes, I had in mind the visualization of the contigs_against_ref.paf 

Oh! Yes, we are thinking in the same direction. I am also going to visualize SV from Ragoo's output. My plan is to align Nanopore reads to ragoo.fasta then visualize the ragoo.fasta.bam using Ribbon.
What do you think about this plan? Is there any other way to visualize SVs?

**Comment by malonge** [comment]
Hi there,

To answer your question more specifically, I wonder what relationship you want to study. It sounds like you want to use ragoo.fasta as a reference genome and use nanopore reads from other samples to call SVs with respect to ragoo.fasta. Is that right? One could also look at SVs in ragoo.fasta with respect to another reference.

**Comment by Ural-Yunusbaev** [comment]
First, I want to visualize the synteny between the target contigs and guide chromosome, then between the target contigs and output chromosome. Like this:

```
Guide  chr  ===============================  ragoo/in/reference.fasta
            |||  ||||||  |||  |  ||||||||||

Target cont ===  ======  ===  =  ==========  ragoo/in/contigs.fasta
            |||  ||||||  |||  |  ||||||||||
Output chr  ===============================  ragoo/out/ragoo.fasta
```

**Comment by hyphaltip** [comment]
Did you try minimap2 with paftools.js ? That will give that type of output of pairwise alignment on large scale.

If you want too see dot plot, D-GENIES works well http://dgenies.toulouse.inra.fr/

Jason Stajich, PhD
jasonstajich.phd@gmail.com

**Comment by mictadlo** [comment]
Hi, I was able to  get Assemblytics running with:

```
> cd ragoo_output/
> gzip pm_alignments/pm_against_ref.sam.delta
Upload the gz file to http://assemblytics.com/
```
However, does RaGOO has any script for [SplitThreader](http://splitthreader.com/)?

Thank you in advance,

Michal

**Comment by Ural-Yunusbaev** [comment]
> ragoo_output/
> > gzip pm_alignments/pm_against_ref.sam.delta

Hi Michal! But how did you obtain pm_against_ref.sam.delta using ragoo?
Thank you in advance,
Ural

**Comment by malonge** [comment]
@mictadlo For SplitThreader, it looks like you need a VCF file and a CNV file. You can convert the assemblytics output format to VCF format [using SURVIVOR](https://github.com/malonge/RaGOO/wiki/Calling-Structural-Variants#output). I am not sure about the CNV file.
